File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    
    for query in queries:
        s, e, k = query
        valid_values = [value for value in arr[s:e+1] if value > k]
        
        if valid_values:
            answer.append(min(valid_values))
        else:
            answer.append(-1)
    return answer

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        Arrays.fill(answer,-1);
        for(int idx=0; idx<queries.length;idx++){
            int s = queries[idx][0];
            int e = queries[idx][1];
            int k = queries[idx][2];
            int minGreaterValue = Integer.MAX_VALUE;
            for(int j=s;j<=e;j++){
                if(arr[j]>k && arr[j]<minGreaterValue){
                    minGreaterValue = arr[j];
                }
            }
            if(minGreaterValue != Integer.MAX_VALUE){
                answer[idx]=minGreaterValue;
            }
            
        }
        return answer;
    }
}